closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -932,7 +931,8 @@ private void initConstraints() {
          * @return true if the Service is allowed
          */
         boolean isRestrictedServiceAllowed(Service service) {
-            String providerName = service.getProvider().getName();
+            Provider provider = service.getProvider();
+            String providerName = provider.getClass().getName();
 
             // Provider with argument, remove argument.
             // e.g. SunPKCS11-NSS-FIPS, remove argument -NSS-FIPS.
```

Comment:
This no longer makes sense (`providerName` is a fully-qualified class name).

Reply:
Yes, because we get the provider class name from the `Service`, so no need to remove argument. I removed these lines. 

And I also removed the "Remove argument" codes from the `initProviders()` method, because we only set the  fully-qualified class name in the java.security profile, so no need to remove the argument.